src/Core/HTTPMiddleware.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace SilverStripe\Core;
+
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\HTTPResponse;
+
+/**
+ * HTTP Request middleware
+ * Based on https://github.com/php-fig/fig-standards/blob/master/proposed/http-middleware/middleware.md#21-psrhttpservermiddlewaremiddlewareinterface
+ */
+interface HTTPMiddleware
```

Comment:
I wonder if this should be part of SilverStripe\Control, since ti depends on HTTPRequest response?